src/__mocks__/MockStorage.ts

```diff
@@ -1,29 +1,37 @@
 import { PersistentStorage } from '../types';
 
-export default class MockStorage implements PersistentStorage {
-  storage: Map<string, string>;
+export default class MockStorage implements PersistentStorage<any> {
+  storage: Map<string, any>;
 
   constructor() {
     this.storage = new Map();
   }
 
-  setItem(key: string, data: string): Promise<any> {
+  setItem(key: string, data: any): Promise<void> {
     return new Promise(resolve => {
       this.storage.set(key, data);
       resolve();
     });
   }
 
-  removeItem(key: string): Promise<any> {
+  removeItem(key: string): Promise<void> {
     return new Promise((resolve, reject) => {
       this.storage.delete(key);
       resolve();
     });
   }
 
-  getItem(key: string): Promise<string> {
+  getItem(key: string): Promise<any> {
     return new Promise((resolve, reject) => {
       resolve(this.storage.get(key));
     });
   }
+
+  getSize(key: string): Promise<number> {
+    return new Promise(resolve => {
+      if (this.storage.size === 0) return resolve(0);
+      const data = this.storage.get(key);
+      resolve((typeof data === 'string' ? data : JSON.stringify(data)).length);
```

Comment:
Worried about performance implication of serialization just for size limitation.
Typically when using non strings clients data can be very large (like in index db)
I would be reluctant to merge it.

Maybe something like Object.keys(data).length could work?

Reply:
I think that would require another sizeFunction argument to be passed by user then default one would simply use string length function